tests/unit/GraphQL/ApiKeyAuthenticatorTest.php

```diff
@@ -0,0 +1,163 @@
+use SilverStripe\Security\Member;
+use Sminnee\ApiKey\MemberApiKey;
+use SilverStripe\Control\HTTPRequest;
+use SilverStripe\Control\Director;
+use Sminnee\ApiKey\GraphQL\ApiKeyAuthenticator;
+use SilverStripe\GraphQL\Auth\Handler;
+use Sminnee\ApiKey\ApiKeyMemberExtension;
+use SilverStripe\Core\Config\Config;
+use Sminnee\ApiKey\ApiKeyRequestMiddlware;
+use SilverStripe\ORM\ValidationException;
+
+class ApiKeyAuthenticatorTest extends SapphireTest
+{
+    /**
+     * @var Member
+     */
+    private $member;
+
+    /**
+     * @var MemberApiKey
+     */
+    private $key;
+
+    /**
+     * @var string
+     */
+    private $headerName;
+
+    /**
+     * @var string
+     */
+    protected static $fixture_file = 'ApiKeyAuthenticatorTest.yml';
+
+    /**
+    * @var array
+    */
+    protected static $required_extensions = [
+        Member::class => [
+            ApiKeyMemberExtension::class,
+        ],
+    ];
+
+    public function setUp()
+    {
+        parent::setUp();
+        $this->member = $this->objFromFixture(Member::class, 'admin');
+        $this->key = MemberApiKey::createKey($this->member->ID);
+        $this->headerName = Config::inst()->get(ApiKeyRequestMiddlware::class, 'header_name');
+        Handler::config()->authenticators = [
+            [
+                'class' => ApiKeyAuthenticator::class,
+                'priority' => 30,
+            ],
+        ];
+    }
+
+    public function testNoKey()
+    {
+        $request = new HTTPRequest('POST', Director::absoluteBaseURL() . '/graphql');
+        $authenticator = new ApiKeyAuthenticator();
+
+        $applicable = $authenticator->isApplicable($request);
+        $this->assertFalse($applicable);
+
+        $this->expectException(ValidationException::class);
+        $this->expectExceptionMessage(
+            'API key header not readable - please check system logs.'
+        );
+        $authenticator->authenticate($request);
+
+        $request = new HTTPRequest('POST', Director::absoluteBaseURL() . '/graphql');
+        $request = $request->addHeader($this->headerName, null);
+        $authenticator = new ApiKeyAuthenticator();
+
+        $this->expectException(ValidationException::class);
+        $this->expectExceptionMessage(
+            'Specified API key is not assigned to any members.'
+        );
+        $authenticator->authenticate($request);
+    }
+
+    public function testInvalidKey()
+    {
+        $request = new HTTPRequest('POST', Director::absoluteBaseURL() . '/graphql');
+        $request = $request->addHeader($this->headerName, 'borkborkbork');
+        $authenticator = new ApiKeyAuthenticator();
+
+        $this->expectException(ValidationException::class);
+        $this->expectExceptionMessage(
+            'Specified API key was not found.'
+        );
+        $authenticator->authenticate($request);
+
+        // dangling key
+        $request = new HTTPRequest('POST', Director::absoluteBaseURL() . '/graphql');
```

Comment:
From here to the rest of the test method looks like it could be its own test method e.g. `testDanglingKey()`